src/internal/storage/fileset/compaction.go

```diff
@@ -128,6 +88,26 @@ func NewDistributedCompactor(s *Storage, maxFanIn int, workerFunc CompactionBatc
 
 // Compact runs a compaction on the ids
 func (c *DistributedCompactor) Compact(ctx context.Context, ids []ID, ttl time.Duration) (*ID, error) {
```

Comment:
We need a comment for this function, because it is not just a distributed version of Storage.Compact anymore.  It's the only thing that actually does the exponential levels.  Also note that it can return a Composite fileset, unlike Storage.Compact

Reply:
Yeah, it probably makes sense to pull out the level based part of this into a separate function / abstraction so that it is split such that the level based, distributed, and regular compaction are clearly divided into separate concepts.

Reply:
Okay so something like:
```go
type Compactor interface {
  Compact(context.Context, ids []ID, ttl time.Duration) (*ID, error)
}

func DelayedCompact(ctx, c Compactor, ids []ID, ttl time.Duration) ([]ID, error) {
  // level based logic here, call c.Compact
}
```

Reply:
This is pretty similar to what I went with, the main difference being that I used a compaction callback function rather than an interface. I don't really have a strong preference for interface vs callback, so if you prefer interface I can switch it over to that.